tests/acceptance/30_custom_promise_types/06_cp_with_classes.cf

```diff
@@ -0,0 +1,85 @@
+######################################################
+#
+#  Basic test of promise module using the shell library and producing classes
+#
+#####################################################
+body common control
+{
+    inputs => { "../default.cf.sub" };
+    bundlesequence  => { default("$(this.promise_filename)") };
+    version => "1.0";
+}
+
+#######################################################
+
+bundle agent init
+{
+  vars:
+    "test_string"
+      string => "hello, modules";
+
+   files:
+    "$(G.testfile).src"
+      create => "true",
+      edit_line => insert_lines("$(test_string)");
+
+    "$(G.testfile).dst"
+      delete => init_delete;
+
+    "$(this.promise_dirname)/cfengine.sh"
+      copy_from => local_cp("$(this.promise_dirname)/../../../docs/custom_promise_types/cfengine.sh");
+}
+
+body delete init_delete
+{
+      dirlinks => "delete";
+      rmdirs   => "true";
+}
+
+#######################################################
+
+promise agent example
```

Comment:
`promise agent cp` (or change the examples in modules to use the `example` promise type)